### DISCUSSION TITLE
Fine-grained token permissions

### DISCUSSION THREAD

**Discussion by OsePedro** [upptime/upptime#700] [Q&A]
I'm trying to replace the personal access token with a [fine-grained personal access token](https://github.blog/2022-10-18-introducing-fine-grained-personal-access-tokens-for-github/), so that I won't have to give the upptime workflows access to all of my repositories. I've attempted to identify the set of fine-grained permissions that correspond to the "repo" and "workflow" scopes that the classic token requires, and it looks like it might be these:

- actions
- code
- commit statuses
- deployments
- issues
- pages
- workflows

I've manually triggered all of the workflows, and they all ran successfully, so it seems that this is a sufficient set of permissions. But can anyone confirm whether or not I need _all_ of these?

**Comment by AnandChowdhary** [comment]
We don't have an official list of these scopes, but if you're able to make everything work with these scopes then we should recommend everyone to switch to a fine-grained PAT. Can you share your findings if everything keeps working correctly after a few days too, just to ensure that scheduled workflows work too?

**Reply by OsePedro** [reply]
It's been 5 days since I switched to the fine-grained personal access token now, and all CI workflows except for "Setup CI" have been automatically scheduled and successfully executed.

**Reply by AnandChowdhary** [reply]
Sounds great, thank you so much!

**Reply by drzraf** [reply]
Read/write for all or are some of them suitable for read-only?

**Reply by AnandChowdhary** [reply]
I think we need some write permissions to push to the repository but maybe for commit statuses we just need read access? I'm not sure, maybe @OsePedro would know better?

**Reply by OsePedro** [reply]
Sorry, I'm new to GitHub access tokens in general, so I'm not sure either. I only became aware that fine-grained tokens exist last week, when I had to renew my previous access token. I wasn't really concerned about what the workflows can do to my upptime repository, I was only concerned about blocking their access to other repositories. So I granted read/write access to all of the above permissions without thinking about it.
